scripts/support/runner/core.py

```diff
@@ -142,15 +98,89 @@ def _parser_darwin(self, cmd):
         return {int(b) for b in used_bytes}
 
 
+class BuildConfiguration:
+    '''This helper class provides access to build-time configuration.
+
+    Configuration options can be read as if the object were a dict,
+    either object['CONFIG_FOO'] or object.get('CONFIG_FOO').
+
+    Configuration values in auto.conf and generated_dts_board.conf are
+    available.'''
+
+    def __init__(self, build_dir):
+        self.build_dir = build_dir
+        self.options = {}
+        self._init()
+
+    def __getitem__(self, item):
+        return self.options[item]
+
+    def get(self, option, *args):
+        return self.options.get(option, *args)
+
+    def _init(self):
+        generated = os.path.join(self.build_dir, 'zephyr', 'include',
+                                 'generated')
+        kconfig = os.path.join(self.build_dir, 'zephyr', 'kconfig', 'include',
+                               'config')
+        files = [os.path.join(generated, 'generated_dts_board.conf'),
+                 os.path.join(kconfig, 'auto.conf')]
```

Comment:
CMake parses values from ${DOTCONFIG} which is located somewhere else than auto.conf.

I don't know what the difference between auto.conf and ${DOTCONFIG} is. But I'm sure
we want CONFIG_* vars to be the same in the CMake namespace and in this namespace
so we should set them to the same file.

Reply:
Oh, OK, nice catch! That's a good point.